src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -76,12 +76,19 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    //Use JSoup to sanitize the user data
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
     //Get value of a query parameter
     String recipient = request.getParameter("recipient");
 
+    //Add regular expression replacement logic
+    //Replace img URLs with <img> elements
+    String regex = "(https?://\\S+\\.(png|jpg))";
```

Comment:
What about supporting GIFs (and also "jpeg", which is an alternative extension representing JPEG files)?